mautic/mautic
# Import validation

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

This PR adds validation to required fields during the import process. Imports will fail to be created and notify the user that they must map all required fields before allowing them to create the import. It also validates that a file was selected and warns users that they are submitting the upload form without a file.

#### Steps to test this PR:
1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Make a required custom field
3. Import a CSV file without this required field - it not validate the mapping form.
3. With this PR the import form will fail to submit until you map all required fields.


<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10884"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
